classes/statistics/MetricsDAO.inc.php

```diff
@@ -94,9 +94,241 @@ protected function getAssocObjectInfo($submissionId, $contextId) {
+							THEN 0
+						END
+					) AS submission_declined_other,
+					COUNT(CASE WHEN s.status = ' . STATUS_DECLINED . ' THEN 0 END) AS submission_declined_total,
+				
+					-- average days to decide
+					AVG(
+						CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_ACCEPT . '
+							THEN DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))
+						END
+					) AS submission_days_to_accept,
+					AVG(
+						CASE WHEN ed.decision IN (' . SUBMISSION_EDITOR_DECISION_DECLINE . ', ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ')
+							THEN DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))
+						END
+					) AS submission_days_to_reject,
+					AVG(DATEDIFF(first_ed.date_decided, COALESCE(s.date_submitted, s.last_modified))) AS submission_days_to_first_decide,
+					AVG(DATEDIFF(ed.date_decided, COALESCE(s.date_submitted, s.last_modified))) AS submission_days_to_decide,
+				
+					-- acceptance/rejection rate
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_ACCEPT . ' THEN 0 END) / COUNT(0) * 100 AS submission_acceptance_rate,
+					COUNT(CASE WHEN s.status = ' . STATUS_DECLINED . ' THEN 0 END) / COUNT(0) * 100 AS submission_rejection_rate,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ' THEN 0 END) / COUNT(0) * 100 AS submission_declined_initial_rate,
+					COUNT(CASE WHEN ed.decision = ' . SUBMISSION_EDITOR_DECISION_DECLINE . ' THEN 0 END) / COUNT(0) * 100 AS submission_declined_rate,
+					COUNT(
+						CASE WHEN s.status = ' . STATUS_DECLINED . '
+							AND (
+								ed.decision NOT IN (' . SUBMISSION_EDITOR_DECISION_DECLINE . ', ' . SUBMISSION_EDITOR_DECISION_INITIAL_DECLINE . ')
+								OR ed.decision IS NULL
+							)
+							THEN 0
+						END
+					) / COUNT(0) * 100 AS submission_declined_other_rate
+				FROM submissions s
+				LEFT JOIN edit_decisions first_ed
+					ON first_ed.edit_decision_id = (
+						SELECT ed.edit_decision_id
+						FROM edit_decisions ed
+						WHERE
+							ed.submission_id = s.submission_id
+						ORDER BY
+							ed.stage_id ASC, ed.round ASC, ed.date_decided ASC
+						LIMIT 1
+					)
+				LEFT JOIN edit_decisions ed
+					ON ed.edit_decision_id = (
+						SELECT ed.edit_decision_id
+						FROM edit_decisions ed
+						WHERE
+							ed.submission_id = s.submission_id
+						ORDER BY
+							ed.stage_id DESC, ed.round DESC, ed.date_decided DESC
+						LIMIT 1
+					)
+				WHERE
+					-- skip unfinished submissions
+					s.submission_progress = 0
+					' . ($where ? ' AND ' . implode(' AND ', $where) : '') . '
+			) AS statistics
+			INNER JOIN (
+				SELECT
+					YEAR(COALESCE(?, CURRENT_TIMESTAMP))
+					- YEAR(
+						COALESCE(
+							?,
+							(
+								SELECT s.date_submitted
+								FROM submissions s
+								WHERE s.date_submitted IS NOT NULL
+								ORDER BY s.date_submitted
+								LIMIT 1
+							),
+							CURRENT_TIMESTAMP
+						)
+					) + 1 AS count
+			) AS years
```

Comment:
Does this work on PostgreSQL?

Reply:
I just checked against MySQL. To port specifically this part against PostgreSQL, replacing YEAR by EXTRACT will suffice.